crates/transaction-pool/src/pool/mod.rs

```diff
@@ -731,6 +731,14 @@ where
         self.get_pool_data().all().transactions_iter().filter(|tx| tx.origin == origin).collect()
     }
 
+    /// Returns all pending transactions filted by [`TransactionOrigin`]
+    pub(crate) fn get_pending_transactions_by_origin(
+        &self,
+        origin: TransactionOrigin,
+    ) -> Vec<Arc<ValidPoolTransaction<T::Transaction>>> {
+        self.pending_transactions().iter().filter(|tx| tx.origin == origin).cloned().collect()
```

Comment:
you can use into_iter here and avoid the clone, if you go down one more level we should be able to avoid the clone entirely and filter directly on the subpool level

Reply:
When do you say “down one more level” is referring to use into_iter?